src/main/java/org/myrobotlab/service/NeoPixel.java

```diff
@@ -26,501 +26,829 @@
+    return pin;
   }
 
-  public void attach(NeoPixelController controller, int pin, int numPixel, int depth) {
-    if (controller == null) {
-      error("setting null as controller");
-      return;
+  public int getPixelDepth() {
+    return pixelDepth;
+  }
+
+  public PixelSet getPixelSet() {
+    return getPixelSet(null, null);
+  }
+
+  /**
+   * Get the pixel set requested - if it does not exist and the pixel set index
+   * is off by one it will create a new pixel set of 0,0,0,0 value add it to the
+   * matrix and return it
+   * 
+   * @param matrixName
+   * @param pixelSetIndex
+   * @return
+   */
+  public PixelSet getPixelSet(String matrixName, Integer pixelSetIndex) {
+    if (matrixName == null) {
+      matrixName = currentMatrix;
     }
-    if (isAttached) {
-      log.info("Neopixel already attached");
-      return;
+
+    if (pixelSetIndex == null) {
+      pixelSetIndex = currentSequence;
     }
 
-    this.pin = pin;
-    this.numPixel = numPixel;
-    this.depth = depth;
+    List<PixelSet> pixelSets = matrices.get(matrixName);
 
-    // clear the old matrix
-    pixelMatrix.clear();
+    if (pixelSets == null) {
+      // make new matrix
+      pixelSets = new ArrayList<>();
+      matrices.put(matrixName, pixelSets);
+    }
 
-    // create a new matrix
-    for (int i = 1; i < numPixel + 1; i++) {
-      setPixel(new PixelColor(i, 0, 0, 0, 0));
+    // add new pixel set if we dont have the one requested
+    if (pixelSetIndex > pixelSets.size()) {
+      error("sequence %d out of bounds", pixelSetIndex);
+      return null;
     }
 
-    controller.neoPixelAttach(this, pin, numPixel);
+    // sequence address == size we need a pixel set created
+    if (pixelSetIndex == pixelSets.size()) {
+      PixelSet ps = new PixelSet();
+      for (int i = 0; i < pixelCount; ++i) {
+        ps.pixels.add(new Pixel(i, 0, 0, 0, 0));
+      }
+      pixelSets.add(ps);
+    }
+    return pixelSets.get(pixelSetIndex);
+  }
 
-    log.info("{} setController {}", getName(), controller.getName());
-    this.controller = controller;
-    controllerName = this.controller.getName();
-    isAttached = true;
-    // update gui with full pixels
-    writeMatrix();
-    broadcastState();
+  public int getRed() {
+    return red;
   }
 
-  @Override
-  public void detach(NeoPixelController controller) {
-    // let the controller you want to detach this device
-    if (controller != null) {
-      controller.detach(this);
-    }
-    // setting controller reference to null
-    this.controller = null;
-    isAttached = false;
+  // @Override
+  public void onEndSpeaking(String utterance) {
+    clear();
+  }
+
+  // TODO - onStarted
+  public void onStarted(String name) {
     refreshControllers();
-    broadcastState();
   }
 
-  public void refresh() {
+  // @Override
+  public String onStartSpeaking(String utterance) {
+    startAnimation();
+    return utterance;
+  }
+
+  public void playAnimation(String animation) {

```

Comment:
These are all implemented on the UC.. why do we have them here?  This seems like it's more like debug/test code, rather than code that should be running here.

Reply:
this is to play any animation - uC or Java-land both with be available be played by name here